eth/gasprice/gasprice.go

```diff
@@ -213,26 +208,12 @@ func NewOracle(backend OracleBackend, config Config) (*Oracle, error) {
 // produced at the current time. If ApricotPhase3 has not been activated, it may
 // return a nil value and a nil error.
 func (oracle *Oracle) EstimateBaseFee(ctx context.Context) (*big.Int, error) {
-	_, baseFee, err := oracle.suggestDynamicFees(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	// We calculate the [nextBaseFee] if a block were to be produced immediately.
-	// If [nextBaseFee] is lower than the estimate from sampling, then we return it
-	// to prevent returning an incorrectly high fee when the network is quiescent.
 	nextBaseFee, err := oracle.estimateNextBaseFee(ctx)
-	if err != nil {
-		log.Warn("failed to estimate next base fee", "err", err)
-		return baseFee, nil
-	}
-	// If base fees have not been enabled, return a nil value.
 	if nextBaseFee == nil {
```

Comment:
I think we can have `err != nil` but `nextBase == nil` here and in that case we won't return the err. I wonder if we should just `oracle.estimateNextBaseFee(ctx)` directly.

Reply:
I think this function can just be `return oracle.estimateNextBaseFee(ctx)`. `estimateNextBaseFee` returns `nil` if the `BaseFee` on the last accepted block is `nil`, which guarantees that dynamic fees has not only been scheduled, but is already activated.

Reply:
I think this won't ever return it (since we already past AP3). if you want to maintain the old behaviour then we should log.warn the err and then `return nil,nil`, but I think we can return the err as well.